### PR TITLE
Improvements to Vulnerability Data Export functionality.

### PR DESCRIPTION
**Pull request by prahaladdarkin**
Closes:
* https://github.com/goharbor/harbor/issues/17152
* https://github.com/goharbor/harbor/issues/17153
Better error handling in case of task executions not found in the system

Signed-off-by: prahaladdarkin <prahaladd@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by AllForNothing** [comment]
@prahaladdarkin As you changed swagger.yml,   so we also need to update the UI code, otherwise, you can never pass the UI-UT . I've pushed a new commit to your main branch.   You can  use  git rebase to squash my  commit  into yours. it can help to pass the DCO check

File: src/server/v2.0/handler/scanexport.go [hunk 1]
```diff
@@ -131,8 +131,21 @@ func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operat
 		return se.SendError(ctx, err)
 	}
 	execution, err := se.scanDataExportCtl.GetExecution(ctx, params.ExecutionID)
+
 	if err != nil {
-		se.SendError(ctx, err)
+		if notFound := orm.AsNotFoundError(err, "execution with id: %d not found", params.ExecutionID); notFound != nil {
+			return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
+				writer.WriteHeader(http.StatusNotFound)
```
Comment (wy65701436) [inline src/server/v2.0/handler/scanexport.go:138]: return errors.New(nil).WithCode(errors.NotFoundCode).WithMessage("... not found")

File: src/server/v2.0/handler/scanexport.go [hunk 2]
```diff
@@ -131,8 +131,21 @@ func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operat
...
+			})
+		}
+		return se.SendError(ctx, err)
+	}
+
+	// check if the CSV artifact for the execution exists
+	if !execution.FilePresent {
+		return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
```
Comment (wy65701436) [inline src/server/v2.0/handler/scanexport.go:146]: ditto

File: src/server/v2.0/handler/scanexport.go [hunk 3]
```diff
@@ -143,7 +156,7 @@ func (se *scanDataExportAPI) DownloadScanData(ctx context.Context, params operat
 
 	if secContext.GetUsername() != execution.UserName {
 		return middleware.ResponderFunc(func(writer http.ResponseWriter, producer runtime.Producer) {
-			writer.WriteHeader(http.StatusUnauthorized)
+			writer.WriteHeader(http.StatusForbidden)
```
Comment (wy65701436) [inline src/server/v2.0/handler/scanexport.go:159]: return errors.New(nil).WithCode(errors.ForbiddenCode).WithMessage("...")

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/server/v2.0/handler/scanexport.go [hunk 4]
```diff
@@ -173,7 +186,12 @@ func (se *scanDataExportAPI) GetScanDataExportExecutionList(ctx context.Context,
 	if err != nil {
 		return se.SendError(ctx, err)
 	}
-	executions, err := se.scanDataExportCtl.ListExecutions(ctx, params.UserName)
+	secContext, err := se.GetSecurityContext(ctx)
+
```
Comment (chlins) [inline src/server/v2.0/handler/scanexport.go:190]: move this empty line after current line 193.

**Review by chlins — APPROVED** [review]
lgtm

**Pull request merged**